src/bin/main.rs

```diff
@@ -42,6 +42,7 @@ fn main() {
 
 /// Rustfmt operations.
 enum Operation {
+    Failure(String),
```

Comment:
I think that adding this variant to `Operation` is fine since it creates a minimal diff. On the other hand, I also feel that this should be part of some kind of error type.

AFAICT the return type of `determine_operation` does not need to be `ErrorKind` (it just relies on the fact that `ErrorKind` implements `From<std::io::Error>`), so what about adding a new error type that represents the failure of `determine_operation`? I am thinking of something like the following enum:

```rust
// +1 if this derives `Fail`.
enum OperationError {
    UnknownHelpTopic(String),
    UnknownPrintConfigTopic(String),
    MinimalPathWithStdin,
}
```

Please feel free to bikeshed on the naming.

Reply:
sounds good!